src/lib/settings.js

```diff
@@ -0,0 +1,359 @@
+		_.extend(this.settings, this.overrides);
+
+		const filename = this.findOverridesFile(profile);
+		try {
+			fs.writeFileSync(filename, JSON.stringify(this.overrides, null, 2), { mode: '600' });
+		} catch (ex) {
+			console.error('There was an error writing ' + filename + ': ', ex);
+		}
+	}
+
+	transitionSparkProfiles(fs=require('fs'), notifyTranslate) {
+		const sparkDir = path.join(this.findHomePath(), '.spark');
+		const particleDir = path.join(this.findHomePath(), '.particle');
+		if (fs.existsSync(sparkDir) && !fs.existsSync(particleDir)) {
+			fs.mkdirSync(particleDir);
+
+			if (notifyTranslate) {
+				notifyTranslate(sparkDir, particleDir);
+			}
+
+			const files = fs.readdirSync(sparkDir);
+			files.forEach((filename) => {
+				const data = fs.readFileSync(path.join(sparkDir, filename));
+				let jsonData;
+				try {
+					jsonData = JSON.parse(data);
+				} catch (ex) {
+					// invalid JSON, don't transition
+					return;
+				}
+
+				if (filename === 'profile.json') {
+					if (jsonData.name === 'spark') {
+						jsonData.name = 'particle';
+					}
+				}
+
+				if (filename === 'spark.config.json') {
+					filename = 'particle.config.json';
+				}
+
+				if (jsonData.apiUrl && jsonData.apiUrl.indexOf('.spark.io') > 0) {
+					jsonData.apiUrl = jsonData.apiUrl.replace('.spark.io', '.particle.io');
+				}
+
+				const jsonString = JSON.stringify(jsonData, null, 2);
+				fs.writeFileSync(path.join(particleDir, filename), jsonString, { mode: '600' });
+			});
+		}
+	}
+
+	listConfigs(particleDir = this.ensureFolder()) {
+		const files = utilities.globList(null, [
+			path.join(particleDir, '*.config.json')
+		]);
+
+		return files.map(item => {
+			const filename = path.basename(item);
+			//strip the extension
+			const name = filename.replace('.config.json', '');
+			return name;
+		});
+	}
+
+	/**
+	 * Retrieves a function that can retrieve or set a value.
+	 * @param {string} key       The key of the setting to fetch/update
+	 * @returns {function(newValue)}   A function that fetches the current value, or updates if an argument
+	 * is defined.
+	 */
+	fetchUpdate(key) {
+		function fetchUpdateKey(key, value) {
+			if (value!==undefined) {
+				this.set(key, value);
+				return value;
+			} else {
+				return this.get(key);
+			}
+		}
+		return fetchUpdateKey.bind(this, key);
+	}
+}
+
+const defaultSettings = {
+	apiUrl: 'https://api.particle.io',
+	buildUrl: 'https://build.particle.io',
+	access_token: null,
+
+	notSourceExtensions: [
+		'.ds_store',
+		'.jpg',
+		'.gif',
+		'.png',
+		'.include',
+		'.ignore',
+		'.ds_store',
+		'.git',
+		'.bin'
+	],
+
+	dirIncludeFilename: 'particle.include',
+	dirExcludeFilename: 'particle.ignore',
+
+	knownApps: {
+		'deep_update_2014_06': true,
+		'cc3000': true,
+		'cc3000_1_14': true,
+		'tinker': true,
+		'voodoo': true
+	},
+	knownPlatforms: {
+		0: 'Core',
+		6: 'Photon',
+		8: 'P1',
+		10: 'Electron',
+		88: 'Duo',
+		103: 'Bluz'
+	},
+	updates: {
```

Comment:
the thought was that maybe want to also have updates in Dev...I suppose eventually all the update flies and platform IDs will move to yet another module that specializes in accessing firmware for each platform (tinker, system firmware etc...)